src/ghost_cursor.rs

```diff
@@ -17,7 +17,7 @@
 //! This is the crux of the issue, and the most likely place for unsoundness in the whole scheme.
 //!
 //! Let us start by a broken example to better understand what we are looking for. Let us imagine a simple doubly linked
-//! list data-structure where each node has an optional previous and next fields to point to the previous and next nodes
+//! list data structure where each node has an optional previous and next field to point to the previous and next node,
```

Comment:
There are two fields (previous and next), hence I believe the plural is correct here.

Reply:
I'm pretty sure my formulation is correct, each node has only a single optional previous field and a single optional next field. The previous formulation is definitely wrong though, "an" is for singular.

Reply:
The previous formulation was definitely mismatching indeed.

I asked for help on Stack Overflow (https://ell.stackexchange.com/questions/314926/is-a-noun-used-for-multiple-adjectives-singular-or-plural), and they pointed out that it's not clear if there's a single optional field (for both previous and next at once) or two with the current formulation.

Instead, the suggestion is to make it clearer there are two fields:

> list data structure where each node has two optional fields, a previous and a next field, to point respectively to the previous and next node,

What do you think?

Reply:
That's a good point, I will use your suggestion.